### PR TITLE
Add patiencediff python library

### PR DESCRIPTION
**Pull request by wildsebastian**
Add external patiencediff library to python-modules and as a dependency
to breezy.

###### Motivation for this change
nix-prefetch-baz does not build with latest breezy version 3.1.0. This requires the external package `patiencediff`.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/patiencediff/default.nix [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    sha256 = "0nm2242xgg59l06m54rzxp41aly3lxjh0a1s7h6dk7ryxjh002lv";
+  };
+
+  # too complicated to setup
+  doCheck = false;
+
+  meta = with stdenv.lib; {
+    description = "This package contains the implementation of the patiencediff algorithm";
+    homepage = "https://www.breezy-vcs.org/";
+  };
```
Comment (nyanloutre) [inline pkgs/development/python-modules/patiencediff/default.nix:18]: Could you add the license and a maintainer ?

**Review by nyanloutre — APPROVED** [review]
Tested it successfully on my system

File: pkgs/development/python-modules/patiencediff/default.nix (hunk 1 above)
Comment (wildsebastian) [reply pkgs/development/python-modules/patiencediff/default.nix:18]: Sure thing!

Comment (wildsebastian) [reply pkgs/development/python-modules/patiencediff/default.nix]: I hope it is fine that I added myself as maintainer. I did not want to push it to someone else.

**Comment by orivej** [comment]
Thank you! I have made some changes to this PR:
- split into multiple commits; (The general rule is to have one commit per package or per a non-package directory.)
- named them according to the [CONTRIBUTING](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md) guide;
- enabled patiencediff tests; (Did you copy `doCheck = false; # too complicated to setup` from another package?)
- changed its license from gpl2 to gpl2Plus. (Its source say "either version 2 of the License, or (at your option) any later version".)

**Comment by wildsebastian** [comment]
@orivej Thank you!
- Sorry, I missed the multi commit rule.
- Also my bad, sorry for that.
- My bad again
Thank you for your comment! I will be more careful next time!

**Pull request merged**